query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/ViewDefinition.java

```diff
@@ -120,6 +128,11 @@ public static ViewDefinition parse(Config config, String tenantColumnName) {
         spec.addColumnName(physName);
         spec.setType(ValueType.STRING);
       }
+
+      if (textIndexFields.contains(physName)) {
+        spec.setTextIndex();
```

Comment:
We are maintaining map of logical column names to PinotColumnSpec. The physical columnName is already added above before this lines #128, #129 as last `else` block.

Reply:
ah got it, the code fold got me again! The spec here is a column spec (while I had thought it was a spec for the whole table, since we were adding rather than setting a column name on line 128)